static/js/settings_user_groups.js

```diff
@@ -255,6 +208,71 @@ exports.populate_user_groups = function () {
     });
 };
 
+exports.set_raw_description = function (target, destination) {
+    var group_id = $(target).closest('.user-group').attr("id");
+    var group_data = user_groups.get_user_group_from_id(group_id);
+    if (!group_data) {
+        blueslip.error('set_raw_description() fails');
+        return;
+    }
+    destination.text(group_data.description);
+};
+
+exports.change_group_description = function (e) {
+    e.preventDefault();
+    var group_id = $(e.target).closest(".user-group").attr("id");
+    var user_group_status = $('#user-groups #' + group_id + ' .user-group-status');
+    var group_data = user_groups.get_user_group_from_id(group_id);
+    var description = $('#user-groups #' + group_id + ' .description').text().trim();
+
+    if (group_data.description === description) {
+        return;
+    }
+    channel.patch({
+        url: "/json/user_groups/" + group_id,
+        data: {
+            name: group_data.name,
+            description: description,
+        },
+        success: function () {
+            user_group_status.hide();
+        },
+        error: function (xhr) {
+            var errors = JSON.parse(xhr.responseText).msg;
+            xhr.responseText = JSON.stringify({msg: errors});
+            ui_report.error(i18n.t("Failed"), xhr, user_group_status);
+            $('#user-groups #' + group_id + ' .description').text(group_data.name);
+        },
+    });
+};
+exports.change_group_name = function (e) {
+    e.preventDefault();
+    var group_id = $(e.target).closest(".user-group").attr("id");
+    var user_group_status = $('#user-groups #' + group_id + ' .user-group-status');
+    var group_data = user_groups.get_user_group_from_id(group_id);
+    var name = $('#user-groups #' + group_id + ' .name').text().trim();
+
+    if (group_data.name === name) {
+        return;
+    }
+    channel.patch({
+        url: "/json/user_groups/" + group_id,
+        data: {
+            name: name,
+            description: group_data.description,
+        },
+        success: function () {
+            user_group_status.hide();
+        },
+        error: function (xhr) {
+            var errors = JSON.parse(xhr.responseText).msg;
+            xhr.responseText = JSON.stringify({msg: errors});
+            ui_report.error(i18n.t("Failed"), xhr, user_group_status);
+            $('#user-groups #' + group_id + ' .name').text(group_data.name);
+        },
+    });
+};
```

Comment:
There is lots of common code between `change_name` and `change_description` function. You can write common function and then call this function in both different function.

Reply:
will split the code into reusable functions then.